### ISSUE TITLE
VariableDeclarationUsageDistance not able to handle try with resources properly

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#11968]
I have read check documentation: https://checkstyle.org/config_coding.html#VariableDeclarationUsageDistance
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java
```
```
/var/tmp $ cat config.xml
```
```XML
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="VariableDeclarationUsageDistance">
      <property name="validateBetweenScopes" value="true"/>
      <property name="allowedDistance" value="1"/>
    </module>
  </module>
</module>
```
```
/var/tmp $ cat Test.java
```
```java
public class Test {
    void method() {
        String a = ""; // no violation expected
        try (AutoCloseable i = new java.io.StringReader(a)) {
        }
        catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}
```
```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.3.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:3:9: Distance between variable 'a' declaration and its first usage is 2, but allowed 1.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value). [VariableDeclarationUsageDistance]
Audit done.
Checkstyle ends with 1 errors.
```
No error was expected, the variable is used just after it was declared, plus in the source code, there is no mention of `TokenTypes.RESOURCES` so this seems like a false positive.

**Comment by jxr98** [comment]
I am on it.

**Comment by nrmancuso** [comment]
@jxr98 this is a great improvement on this check, we found a few more false positives than we initially thought :)
